### DISCUSSION TITLE
Quick Search 'Letters only' like in TC

### DISCUSSION THREAD

**Discussion by milutinjaric** [doublecmd/doublecmd#1479] [Suggestions]
Currently when letters are typed to locate the file in dir a search dialog appears at the bottom making it impossible to start a new search without closing the dialog first which is annoying if I quickly need to locate multiple files.

I was not able to find such an option in Configuration which would allow me to disable search dialog. Am I missing something or is there a possibility to add such feature?

**Comment by alexx2000** [comment]
No such option.

**Comment by milutinjaric** [comment]
Will it be added?

**Comment by alexx2000** [comment]
Not planned.

**Comment by agh1467** [comment]
I'd like to re-iterate this feature, as it's the file/directory matching selection method I'm most familiar with. Here is some more on this feature to elaborate on how it works in other software:

From Total Commander help documentation:
> Letter only like in Windows Explorer, typing a letter jumps to the next file starting with that letter.
> You can type the first few characters of a file, not only the first letter.
> Total Commander will 'forget' the typed letters after about 1 second delay.

A user post on the forums: https://doublecmd.h1n.ru/viewtopic.php?p=23111
> 1 When pressing a letter:
> To look for an entry that starts with this letter.
> But not to open a search-line that has now this letter written within.
>
> 2a If found and selection moved there:
> If the same letter is pressed again, move to the second entry that also starts with this button.
> Continue and cycle if hitting the last one.
>
> 2b If not found:
> Hitting a new letter would start a new search. Because the first entry was not saved in any text search box.
>
> 3 Helpful feature:
> Above steps 1 and 2 search only based on the first letter of files.
> If hitting quickly (writing speed) two letters, make the second letter a sub search of the first letter.
>
> Example of step 3:
> Quick a + b jumps first to "a.txt" and immediately after to "abacus".
> Slow a + b jumps first to "a.txt" and than to "berta".

I offer other scenarios/examples:

**Typo correction and rapid searching files of different names**
> Typing the first few characters of a file name.
> Make a typo
> wait a second
> Re-type

To correct this and re-search with the current quick search/filter, backspace off the search query or press escape, then re-type the search.

**Rapidly cycling through matching files**
> Typing the string to match on
> Press again the last letter within a second to move to the next matching file.

To do this with current quick search/filter, you have to move to the arrow keys. To do this multiple times it's a lot of back and forth to the escape key/backspace key, and arrow keys.

**Search match and then navigate away from them**

> Typing the string to match on, maybe just a single character like d.
> Use key repetition to navigate within the match set.
> Change your mind on what file you want.
> Use arrow keys to navigate to another file that's outside of the match set, but before or after the match set.

Using the current quick search functionality, navigation outside of the match set requires pressing the escape key first to dismiss the search dialog.

**Search for a file by start with a typo**

> Type a string to match a file, but make a typo with the first character
> Wait a second and then re-type the string but without the typo

Since the quick search do not dismiss a search after a period of time, it's required to press escape or backspace off the typo'd characters, and then re-type the match string.

**Search for a file, and then select that file with space**

> Type a sting to match some files/directories
> Select a file that is in focus using the space key

Since the quick search appears, and remains up, selection of a file is not possible and requires dismissal of the quick search box first, otherwise spaces are included in the quick search criteria.

**Search for a file, and then change directories**

> Type a string looking for a file
> Don't find it in this directory
> Use backspace to go up a directory to go to a different directory

Since the quick search box appears, pressing backspace doesn't go up a directory link it does in the file listing, it removes characters from the search field. You must press escape to dismiss the quick search box. Even backspacing all of the characters out of the quick search box doesn't dismiss the box, you must press escape or click the close button.

To summarize:
This feature is all about quickly and rapidly finding the files/directories that one is looking for _without having to leave the home row_ on the keyboard (no need to navigate with arrow keys).
